File: code/smv-debugger/src/main/java/org/fbme/smvDebugger/panel/condition/Equality.java
package org.fbme.smvDebugger.panel.condition;

import java.util.Map;
import java.util.Objects;

public class Equality extends AbstractEquality {
  public Equality(final Expression first, final Expression second) {
    super(first, second);
  }

  @Override
  public boolean evaluate(final Map<String, String> itemSimpleNameToValueMap) {
    if (!(first instanceof Argument)) {
      return first.evaluate(itemSimpleNameToValueMap) == second.evaluate(itemSimpleNameToValueMap);
    }
    final Argument name = (Argument) first;
    final Argument value = (Argument) second;
    return Objects.equals(itemSimpleNameToValueMap.get(name.getContent()), value.getContent());
  }

  @Override
  public String getSymbol() {
    return "==";
  }
}

File: code/smv-debugger/src/main/kotlin/org/fbme/smvDebugger/panel/condition/Equality.kt
package org.fbme.smvDebugger.panel.condition

class Equality(first: Expression, second: Expression) : AbstractEquality(first, second) {
    override fun evaluate(itemSimpleNameToValueMap: Map<String, String>): Boolean {
        if (first !is Argument) {
            return first.evaluate(itemSimpleNameToValueMap) == second.evaluate(itemSimpleNameToValueMap)
        }
        val value = second as Argument
        return itemSimpleNameToValueMap[first.content] == value.content
    }

    override val symbol: String
        get() = "=="
}
